tests/unit_tests/data_steward/cdr_cleaner/cleaning_rules/remove_ehr_data_past_deactivation_date_test.py

```diff
@@ -0,0 +1,199 @@
+            'suspension_time': ['2018-12-07T08:21:14', '2019-12-07T08:21:14']
+        }
+
+    @mock.patch(
+        'retraction.retract_deactivated_pids.get_date_info_for_pids_tables')
+    @mock.patch(
+        'retraction.retract_deactivated_pids.check_and_create_sandbox_dataset')
+    @mock.patch('retraction.retract_deactivated_pids.get_client')
+    @mock.patch('retraction.retract_deactivated_pids.check_pid_exist')
+    @mock.patch(
+        'utils.participant_summary_requests.get_deactivated_participants')
+    def test_remove_ehr_data_past_deactivation_date(
+        self, mock_get_deactivated_participants, mock_pid_exist, mock_client,
+        mock_check_sandbox, mock_date_info):
+        # Preconditions for participant summary module mocks
+        deactivated_participants_df = pandas.DataFrame(
+            columns=self.columns, data=self.deactivated_participants_data)
+        mock_get_deactivated_participants.return_value = deactivated_participants_df
+
+        # Preconditions for retraction module mocks
+        d = {
+            'project_id': [
+                self.project_id, self.project_id, self.project_id,
+                self.project_id, self.project_id, self.project_id
+            ],
+            'dataset_id': [
+                self.pids_dataset_id, self.pids_dataset_id,
+                self.pids_dataset_id, self.pids_dataset_id,
+                self.pids_dataset_id, self.pids_dataset_id
+            ],
+            'table': [
+                'fake_condition_occurrence', 'fake_drug_exposure',
+                'fake_measurement', 'fake_observation',
+                'fake_procedure_occurrence', 'fake_visit_occurrence'
+            ],
+            'date_column': [
+                None, None, 'measurement_date', 'observation_date',
+                'procedure_date', None
+            ],
+            'start_date_column': [
+                'condition_start_date', 'drug_exposure_start_date', None, None,
+                None, 'visit_start_date'
+            ],
+            'end_date_column': [
+                'condition_end_date', 'drug_exposure_end_date', None, None,
+                None, 'visit_end_date'
+            ]
+        }
+        retraction_info = pandas.DataFrame(data=d)
+        mock_date_info.return_value = retraction_info
+
+        client = mock_client.return_value = self.mock_bq_client
+        client.query.return_value.to_dataframe.return_value = deactivated_participants_df
+
+        mock_pid_exist.return_value = 1
+        sandbox = mock_check_sandbox.return_value = 'foo_pid_dataset_id_sandbox'
+
+        # test
+        returned_queries = red.remove_ehr_data_queries(self.project_id,
+                                                       self.ticket_number,
+                                                       self.pids_project_id,
+                                                       self.pids_dataset_id,
+                                                       self.tablename)
```

Comment:
Sorry one more thing-- should this be `self.pids_table`?

Reply:
should `self.tablename` be `self.pids_table`? If that's the question then it's the same thing just with a different name.